extensions/sql-database-projects/src/models/project.ts

```diff
@@ -823,6 +823,55 @@ export class Project implements ISqlProject {
 		await this.addToProjFile(sqlCmdVariableEntry);
 	}
 
+	/**
+	 * Adds DatabaseSource property to the project
+	 * @param databaseSource Source of the database to add
+	 */
+	public async addDatabaseSourceToProjFile(databaseSource: string): Promise<void> {
+		await this.addValueToProjectProperty(constants.DatabaseSource, databaseSource);
+	}
+
+	/**
+	* Removes DatabaseSource property from the project
+	* @param databaseSource Source of the database to remove
+	*/
+	public async removeDatabaseSourceFromProjFile(databaseSource: string): Promise<void> {
+		await this.removeValueFromProjectProperty(constants.DatabaseSource, databaseSource);
+	}
+
+	/**
+	 * Get array of all database source values, regardless of if value is 'well-know'
+	 * @returns Array of all database source values, regardless of if value is 'well-know'
+	 */
+	public getRawDatabaseSourceValues(): string[] {
+		const databaseSource = this.evaluateProjectPropertyValue(constants.DatabaseSource);
+
+		if (databaseSource !== undefined) {
+			return databaseSource.split(';');
+		} else {
+			return [];
+		}
+	}
+
+	/**
+	 * Get string containing all well-known database sources, separated by semicolon
+	 * @returns Well-known database sources, separated by semicolon
+	 */
+	public getWellKnownDatabaseSourceString(): string {
```

Comment:
I don't think this should be a responsibility of the project class to determine whta a certain consumer considers known and what is not. Yes, for telemetry we are only going to send what we are confident in, but for logging we may dump all of them. I think `getRawDatabaseSourceValues` should be renamed to `getDatabaseSources` and left as the only public API to retrieve these values. The rest should be done either in a helper method or in the consumers themselves.